platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +102,168 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
+                        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+
+        if (filtersHaveBeenInitialized) {
+            initFilter(securityFilterServiceReference, filterConfig);
+        }
+
+        sortedSecurityFilterServiceReferences.add(securityFilterServiceReference);
+
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("The SecurityFilters list is now");
+            for (ServiceReference<SecurityFilter> filter : sortedSecurityFilterServiceReferences) {
+                LOGGER.debug("\t{} {}",
+                        filter.getProperty(Constants.SERVICE_RANKING),
+                        ServicePropertiesUtils.getStringProperty(filter,
+                                HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+            }
+        }
     }
 
-}
+    private void initFilter(ServiceReference<SecurityFilter> securityFilterServiceReference,
+            FilterConfig filterConfig) throws ServletException {
+        final ServletContext servletContext;
+
+        if (filterConfig == null) {
+            LOGGER.warn(
+                    "DelegateServletFilter was initialized with a null filterConfig. Initializing SecurityFilter {} with null ServletContext",
+                    ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                            HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+            servletContext = null;
+        } else {
+            servletContext = this.filterConfig.getServletContext();
+        }
+
+        getContext().getService(securityFilterServiceReference)
+                .init(new SecurityFilterInitFilterConfig(securityFilterServiceReference,
+                        servletContext));
+
+        LOGGER.debug("Initialized SecurityFilter {}",
+                ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+    }
+
+    public void removeSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference) {
+        sortedSecurityFilterServiceReferences.remove(securityFilterServiceReference);
+        getContext().getService(securityFilterServiceReference)
+                .destroy();
+    }
+
+    /**
+     * This inner class is used to instantiate a {@link FilterConfig} from a
+     * {@link SecurityFilter}'s service properties containing init params and the same
+     * {@link ServletContext} as the {@link DelegateServletFilter}. The {@link FilterConfig} is used
+     * to initialize the {@link SecurityFilter}.
+     * <p>
+     * The logic of this inner class is copied from the
+     * {@link org.ops4j.pax.web.extender.whiteboard} feature.
+     */
+    private class SecurityFilterInitFilterConfig implements FilterConfig {
+
+        private final ServiceReference<SecurityFilter> securityFilterServiceReference;
```

Comment:
You'll need to not hang onto this reference. Pull the name and properties off of it as you go through the constructor.